packages/app-mobile/components/SyncStatusIcon.tsx

```diff
@@ -0,0 +1,197 @@
+import * as React from 'react';
+import { Alert, Animated, Easing, StyleSheet, View } from 'react-native';
+import { ReactElement, useCallback, useEffect, useMemo, useRef } from 'react';
+const Ionicon = require('react-native-vector-icons/Ionicons').default;
+const FontAwesomeIcon = require('react-native-vector-icons/FontAwesome5').default;
+
+import { connect } from 'react-redux';
+import { AnyAction } from 'redux';
+
+import { State } from '@joplin/lib/reducer';
+import { SyncReport } from '@joplin/lib/Synchronizer';
+import CustomButton from './CustomButton';
+import { _ } from '@joplin/lib/locale';
+import { themeStyle } from '@joplin/lib/theme';
+import synchronizeButtonPress from '@joplin/lib/components/shared/synchronizeButtonPress';
+
+type DispatchEventCallback = (eventName: AnyAction)=> void;
+
+interface Props {
+	syncStarted: boolean;
+	syncReport: SyncReport;
+	isOnMobileData: boolean;
+	syncOnlyOverWifi: boolean;
+	syncTarget: number;
+
+	themeId: number;
+	noteLastModifiedTimestamp: number|null;
+	dispatch: DispatchEventCallback;
+}
+
+const SyncStatusIconComponent = (props: Props) => {
+	const styles = useStyles(props.themeId);
+	const syncAnim = useRef(new Animated.Value(0)).current;
+
+	useEffect(() => {
+		if (props.syncStarted) {
+			Animated.loop(Animated.timing(syncAnim, {
+				toValue: 360,
+				duration: 5000,
+				easing: Easing.linear,
+				useNativeDriver: true,
+			})).start();
```

Comment:
Does that get cleaned up when the component is unmounted?

Reply:
Fixed!